File: projects/express-typeorm/src/database/archive/address.ts
import { Column, Entity, JoinColumn, OneToOne } from 'typeorm';

import { AddressType } from './address-type';
import { Location } from './location';

@Entity('Address')
export class Address extends Location {

  @Column()
  line1: string;

  @Column()
  line2: string;

  @Column()
  city: string;

  @Column()
  state: string;

  @Column()
  postalCode: string;

  @Column({
    default: 'Australia'
  })
  country: string;

  @OneToOne(type => AddressType, {
    cascade: true,
  })
  @JoinColumn()
  addressType: AddressType;

}

// https://github.com/typeorm/typeorm/blob/master/docs/entities.md

// Error: OneToOne FOREIGN KEY constraint

File: projects/spring-boot/server/src/main/java/org/serendipity/restapi/model/Address.java
package org.serendipity.restapi.model;

import javax.persistence.Entity;
// import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.MapsId;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Address {
  
  @Id
  private long id;
  
  // @OneToOne(fetch = FetchType.LAZY)
  @OneToOne
  @JoinColumn(name = "locationId")
  @MapsId
  private Location location;
  
  private String name;
  
  private String line1;
  
  private String line2;
  
  private String city;
  
  private String state;
  
  private String postalCode;
  
  private String country;
  
  private String addressType;
  
  @Override
  public boolean equals(Object o) {

    if (this == o)
      return true;

    if (!(o instanceof Address))
      return false;

    Address other = (Address) o;

    return id != 0L && id == other.getId();
  }

  @Override
  public int hashCode() {
    return 31;
  }  

}
